servers/gemini-sub/src/gemini_sub/server.py

```
@@ -47,61 +130,89 @@ async def _get_web_state() -> dict:
+    async with (
+        httpx.AsyncClient(
+            http2=True,
+            cookies=state["cookies"],
+            follow_redirects=True,
+            timeout=timeout,
+            headers=request_headers,
+        ) as client,
+        client.stream("POST", url, params=params, data=request_data) as response,
+    ):
+        if response.status_code != 200:
+            raise RuntimeError(f"Gemini RPC failed: {response.status_code}")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)